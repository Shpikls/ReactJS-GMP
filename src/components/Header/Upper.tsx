import { Logo } from '@components/Logo'
import { Button } from '@styled/Button'
import { Flex } from '@styled/Flex'
import * as React from 'react'
import { useContext } from 'react'
import styled from 'styled-components'
import { ModalContext } from '~/context'

export const Upper = (): JSX.Element => {
	const [, setModalState] = useContext(ModalContext)

	const handleOpenAddModal = () => {
		setModalState((prevState) => {
			return {
				...prevState,
				...{ addModal: true },
			}
		})
	}

	return (
		<Flex jContent="space-between">
			<Logo />
			<AddButton onClick={handleOpenAddModal}>+ add movie</AddButton>
		</Flex>
	)
}

const AddButton = styled(Button)`
	width: 177px;
	height: 46px;
	background: rgba(96, 96, 96, 0.68);
	font-weight: 600;
	font-size: 20px;
	line-height: 24px;
	color: #f65261;
`
