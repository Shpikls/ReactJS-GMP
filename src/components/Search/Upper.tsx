import { Logo } from '@components/Logo'
import { Button } from '@styled/Button'
import { Flex } from '@styled/Flex'
import * as React from 'react'
import styled from 'styled-components'

export const Upper = (): JSX.Element => (
	<Flex jContent="space-between">
		<Logo />
		<AddButton>+ add movie</AddButton>
	</Flex>
)

const AddButton = styled(Button)`
	width: 177px;
	height: 46px;
	background: rgba(96, 96, 96, 0.68);
	font-weight: 600;
	font-size: 20px;
	line-height: 24px;
	color: #f65261;
`
