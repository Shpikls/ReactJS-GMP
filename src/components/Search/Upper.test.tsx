import React from 'react'
import { render } from '~/test/test-utils'
import { Upper } from './Upper'

describe('Test Upper.tsx', () => {
	test('To match snapshot', () => {
		const { asFragment } = render(<Upper />)

		expect(asFragment()).toMatchSnapshot()
	})
})
