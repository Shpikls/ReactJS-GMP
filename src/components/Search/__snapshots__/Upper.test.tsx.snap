// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Upper.tsx To match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-dlfnuX kcygjN"
  >
    <span
      class="sc-bdfBQB iqkcpy"
    >
      <span>
        netflix
      </span>
      roulette
    </span>
    <button
      class="sc-gsTEea sc-hKgJUU bYHHgn gJqPpG"
    >
      + add movie
    </button>
  </div>
</DocumentFragment>
`;
